docs/errors/E0373.md

```diff
@@ -0,0 +1,31 @@
+# E0373: mistyped strict inequality operator
+
+While `x! == y` is legal TypeScript for a non-null assertion, it may be a typo for a strict inequality, as in `x !== y`:
+
+```javascript
+let x = 17;
+let y = 38;
+if (x! == y) {
+    alert('Not equal!');
+}
+```
+
+To fix the warning, replace `! ==` with ` !==`:
+
+```javascript
+let x = 17;
+let y = 38;
+if (x !== y) {
+  alert("Not equal!");
+}
+```
+
+Alternatively, if a non-null assertion was the intention, put parenthesis around the left hand side:
+
+```javascript
+let x = 17;
+let y = 38;
+if ((x)! == y) {
```

Comment:
I think this example is incorrect. It'll still trigger the warning, right?

Reply:
Ahhhh good point, I was originally thinking this was `(x)! == y` because that is what's being suggested if a non null assertion is intended, this suggestion still is in the diagnostic but does not prevent the warning from triggering, so this example can be removed.